sdb/commands/stacks.py

```diff
@@ -0,0 +1,255 @@
+    # that a symbol came from. As a result, we created the following
+    # function to implement the `-m` functionality. Whenever we filter
+    # by module name, we find the segment in memory where this module
+    # resides and do the matching based on the address of the function
+    # of the current frame.
+    #
+    @staticmethod
+    def find_module_memory_segment(prog: drgn.Program,
+                                   mod_name: str) -> Tuple[int, int]:
+        """
+        Looks for the segment in memory where `mod_name` is
+        loaded within `prog`.
+
+        Returns:
+            (<base_offset>, <size>) if `mod_name` is found.
+            (-1, 0) otherwise.
+        """
+        for mod in list_for_each_entry('struct module',
+                                       prog['modules'].address_of_(), 'list'):
+            if mod.name.string_().decode("utf-8") == mod_name:
+                return (mod.core_layout.base.value_(),
+                        mod.core_layout.size.value_())
+        return (-1, 0)
+
+    def validate_args(self, prog: drgn.Program,
+                      args: argparse.Namespace) -> None:
+        if args.function:
+            try:
+                func = prog[args.function]
+            except KeyError:
+                raise sdb.CommandError(
+                    self.name,
+                    "symbol '{:s}' does not exist".format(args.function))
+            if func.type_.kind != drgn.TypeKind.FUNCTION:
+                raise sdb.CommandError(
+                    self.name, "'{:s}' is not a function".format(args.function))
+
+        task_states = Stacks.TASK_STATES.values()
+        if args.tstate and not args.tstate in task_states:
+            raise sdb.CommandError(
+                self.name,
+                "'{:s}' is not a valid task state (acceptable states: {:s})".
+                format(args.tstate, ", ".join(task_states)))
+
+        if args.module and Stacks.find_module_memory_segment(
+                self.prog, args.module)[0] == -1:
+            raise sdb.CommandError(
+                self.name,
+                "module '{:s}' doesn't exist or isn't currently loaded".format(
+                    args.module))
+
+    def call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
+        # pylint: disable=too-many-locals
+        # pylint: disable=too-many-branches
+
+        #
+        # As the exception explains the code that follows this statement only
+        # works for linux kernel targets (crash dumps or live systems). When
+        # support for userland is added we can factor the kernel code that
+        # follows into its own function and switch to the correct codepath
+        # depending on the target.
+        #
+        if not self.prog.flags & drgn.ProgramFlags.IS_LINUX_KERNEL:
+            raise sdb.CommandError(self.name,
+                                   "userland targets are not supported yet")
+        self.validate_args(self.prog, self.args)
```

Comment:
Ideally, would we do this validation immediately after we parse the arguments?

This makes me think that maybe we should create another "super class" method, similar to "_init_argparse" that would automatically be called by the super class constructor. This way, we would only need to define the function, and wouldn't have to worry about explicitly calling it here.

I'm OK with this approach for now, but it seems like a general design pattern that might be useful for other commands as well.